### PR TITLE
acl: change authmethod.Validator to take a logger

### PR DESCRIPTION
**Pull request by rboyer**

### CODE REVIEW

File: agent/consul/authmethod/authmethods.go [hunk 1]
```diff
@@ -140,7 +141,9 @@ func NewValidator(method *structs.ACLAuthMethod) (Validator, error) {
 		return nil, fmt.Errorf("no auth method registered with type: %s", method.Type)
 	}
 
-	return factory(method)
+	logger = logger.Named("authmethod").With("authMethodType", method.Type, "authMethodName", method.Name)
```
Comment (rboyer) [inline agent/consul/authmethod/authmethods.go:144]:
```suggestion
	logger = logger.Named("authmethod").With("type", method.Type, "name", method.Name)
```
I kinda like this better

Comment (s-christoff) [reply agent/consul/authmethod/authmethods.go:144]: i also like this better

File: agent/consul/authmethod/kubeauth/k8s.go [hunk 2]
```diff
@@ -20,7 +21,7 @@ import (
 
 func init() {
 	// register this as an available auth method type
-	authmethod.Register("kubernetes", func(method *structs.ACLAuthMethod) (authmethod.Validator, error) {
+	authmethod.Register("kubernetes", func(_ hclog.Logger, method *structs.ACLAuthMethod) (authmethod.Validator, error) {
```
Comment (s-christoff) [inline agent/consul/authmethod/kubeauth/k8s.go:24]: can you eli5 what is happening here? i am confused

Comment (s-christoff) [reply agent/consul/authmethod/kubeauth/k8s.go:24]: nevermind i looked down disregard

Comment (rboyer) [reply agent/consul/authmethod/kubeauth/k8s.go:24]: `_` is the blank identifier. Just like `_, err := blah()`. The body of the function doesn't use the argument.
